### PR TITLE
Use correct map key when deleting endpoints in sanitizer

### PR DESCRIPTION
**Pull request by elcasteel**
# Description
In `upstream_removing_sanitizer` we delete the clusters and endpoints that are associated with upstreams that have errors. We were using the cluster name as the key into both the cluster maps and the endpoint maps. However, the endpoint maps use a different value for the key so these were not being deleted. 

# Context

Users ran into this bug when invalid upstreams were accepted and caused gloo to fail. 

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/5022

### CODE REVIEW

File: projects/gloo/pkg/syncer/sanitizer/upstream_removing_sanitizer.go [hunk 1]
```diff
@@ -46,16 +46,30 @@ func (s *UpstreamRemovingSanitizer) SanitizeSnapshot(
...
 	var removed int64
 
 	// Find all the errored upstreams and remove them from the xDS snapshot
 	for _, up := range glooSnapshot.Upstreams.AsInputResources() {
+
 		if reports[up].Errors != nil {
+
 			clusterName := translator.UpstreamToClusterName(up.GetMetadata().Ref())
+			endpointName := resource.GetResourceName(clusters.Items[clusterName].ResourceProto())
+			switch v := clusters.Items[clusterName].ResourceProto().(type) {
```
Comment (elcasteel) [inline projects/gloo/pkg/syncer/sanitizer/upstream_removing_sanitizer.go:58]: I copied this from the code we use to check the consistency of the snapshot: https://github.com/solo-io/solo-kit/blob/master/pkg/api/v1/control-plane/resource/envoy_resource.go#L110
I don't know how to actually convert from a `ResourceProto` to a specific type so I did this to at least get something to show.

File: projects/gloo/pkg/syncer/sanitizer/upstream_removing_sanitizer.go [hunk 2]
```diff
@@ -74,6 +85,7 @@ func (s *UpstreamRemovingSanitizer) SanitizeSnapshot(
 
 	// If the snapshot is not consistent,
 	if xdsSnapshot.Consistent() != nil {
+		fmt.Printf("consistent %s", xdsSnapshot.Consistent())
```
Comment (kdorosh) [inline projects/gloo/pkg/syncer/sanitizer/upstream_removing_sanitizer.go:88]: debug logging?

**Review by kdorosh — COMMENTED** [review]
nice catch, lgtm overall

File: projects/gloo/pkg/syncer/sanitizer/upstream_removing_sanitizer.go [hunk 3]
```diff
@@ -46,16 +46,27 @@ func (s *UpstreamRemovingSanitizer) SanitizeSnapshot(
...
+					endpointName = cluster.GetEdsClusterConfig().GetServiceName()
+				} else {
+					endpointName = cluster.GetName()
+				}
+			}
+			contextutils.LoggerFrom(ctx).Info("deleting endpoint ", endpointName)
 			// remove cluster and endpoints
 			delete(clusters.Items, clusterName)
-			delete(endpoints.Items, clusterName)
+			delete(endpoints.Items, endpointName)
```
Comment (jenshu) [inline projects/gloo/pkg/syncer/sanitizer/upstream_removing_sanitizer.go:69]: do we still want to call this if endpointName is empty?

Comment (elcasteel) [reply projects/gloo/pkg/syncer/sanitizer/upstream_removing_sanitizer.go:69]: Yeah, empty is probably the wrong default, I'll use the clusterName (old behavior) as a default.

**Review by jenshu — APPROVED** [review]

**Pull request merged**